src/main/java/io/github/ryotan/code/CodeEnum.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * コード値を表すインターフェースです。
+ *
+ * <h3>実装に関する規約</h3>
+ * <p>
+ * このインターフェースを実装するクラスは、以下の規約に従ってください。
+ * </p>
+ * <ul>
+ * <li>enumとして宣言されること</li>
+ * <li>型パラメータ{@code <C>}に自分自身の型を設定すること。</li>
+ * </ul>
+ * <p>実装クラスの例は次のようになります。</p>
+ * <pre><code class="java">
+ * public enum Gender implements CodeEnum&lt;Gender&gt; {
+ *     NOT_APPLICABLE("9", "適用不能"),
+ *     FEMALE("2", "女性"),
+ *     MALE("1", "男性"),
+ *     NOT_KNOWN("0", "不明");
+ *
+ *     private final String value;
+ *     private final String label;
+ *
+ *     Gender(String value, String label, String shortLabel, String english) {
+ *         this.value = value;
+ *         this.label = label;
+ *     }
+ *
+ *     &#64;Override
+ *     public String value() {
+ *         return this.value;
+ *     }
+ *
+ *     &#64;Override
+ *     public String label() {
+ *         return this.label;
+ *     }
+ * }
+ * </code></pre>
+ *
+ * <h3>コードのフィルタリング</h3>
+ * <p>
+ * コードによっては、精査などで使うときに一部の値に絞り込みたい場合もあります。
+ * そういった場合は、{@link Filter}アノテーションを付けたフィールドもしくはメソッドを作成してください。
+ * これらからは、{@link Predicate Predicate&lt;C&gt;}を取得できる必要があります。
+ * </p>
+ * <p>
+ * たとえば、上で例にあげた{@code Gender}を（とても古いシステムと連携しなくてはいけないなどの理由で）JIS
+ * X0303に従ってフィルタリングする必要がある場合には、次のようなフィールドを定義して利用することができます。
+ * </p>
+ * <pre><code class="java">
+ * &#64;Filter
+ * public static final Predicate&lt;Gender&gt; JIS_X0303 = Filters.include(MALE, FEMALE);
+ * </code></pre>
+ * <p>
+ * 定義した{@code JIS_X0303}を利用して、JIS X0303に含まれるコード値のみを取得することが出来ます。
+ * </p>
+ * <pre><code class="java">
+ * Code.values(Gender.class, Gender.JIS_X0303)
+ * </code></pre>
+ *
+ * <h3>コードの別名</h3>
+ * <p>
+ * コードによっては論理名を{@link #label()}以外に設定したい場合もあります。そういった場合は、{@link AliasLabel}アノテーションを利用してください。
+ * {@link AliasLabel}アノテーションはフィールドもしくはメソッドに対して付けることができます。ただし、これらから{@link String}が取得できない
+ * （フィールドの場合は{@link String}型で定義されていない。メソッドの場合は戻り値の方が{@link String}でない）場合は単純に無視されます。
+ * </p>
+ * <p>
+ * 例えば、{@code Gender}に英語の別名を付けたい場合、次のようにして定義することが出来ます。
+ * </p>
+ * <pre><code class="java">
+ * &#64;AliasLabel
+ * public final String ENGLISH_LABEL;
+ * </code></pre>
+ * <p>
+ * 定義されている別名は、{@link Code#alias(Class, String, String)}で取得することが出来ます。このとき、{@code name}パラメータには、
+ * {@link AliasLabel}アノテーションを付与したフィールドあるいはメソッドの名称を設定してください。
+ * </p>
+ *
+ * @param <C> 実装クラス自身の型
+ * @author ryotan
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [At-clauses have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'.](https://www.codacy.com/app/ryotan/code-enum-support/file/2914735243/issues/source?bid=3375378&fileBranchId=3375797#l99)